src/Apps/Order/Routes/Shipping/useAddressAutocomplete.ts

```diff
@@ -0,0 +1,135 @@
+import { getENV } from "Utils/getENV"
+import { useCallback, useState } from "react"
+
+// NOTE: Due to the format of this key (a long string of numbers that cannot be parsed as json)
+// This key must be set in the env as a json string like SMARTY_EMBEDDED_KEY_JSON_JSON='{ "key": "xxxxxxxxxxxxxxxxxx" }'
+const smartyCreds = getENV("SMARTY_EMBEDDED_KEY_JSON")
+console.log({ smartyCreds })
+const key = smartyCreds.key
+
+export type SuggestionJSON = {
+  city: string
+  entries: number
+  secondary: string
+  state: string
+  street_line: string
+  zipcode: string
+}
+
+export interface AddressSuggestion extends AutocompleteInputOptionType {
+  address: Omit<Address, "name">
+  entries: number
+}
+
+export const useAddressAutocomplete = () => {
+  const [result, setResult] = useState<SuggestionJSON[]>([])
+
+  // Add feature flag checks including country, etc here later
+  // console.log({ key, kind: typeof key })
+  const enabled = !!key
+
+  const fetchSuggestions = useCallback(
+    async (searchParam: string, selectedParam?: string) => {
+      console.log({ key })
+      if (!key) return null
+      let url = `https://us-autocomplete-pro.api.smarty.com/lookup?key=${encodeURIComponent(
+        key
+      )}&prefer_ratio=3&search=${encodeURIComponent(searchParam)}`
+
+      if (selectedParam) {
+        url += `&selected=${encodeURIComponent(selectedParam)}`
+      }
+
+      console.log({ url })
+      const response = await fetch(url, {
+        headers: {
+          Host: "us-autocomplete-pro.api.smartystreets.com",
+        },
+      })
+      console.log({ response })
+      const json = await response.json()
+      return json
+    },
+    []
+  )
+
+  const fetchForAutocomplete = useCallback(
+    // these are the parameters to the Smarty API call
+    async (searchParam: string, selectedParam?: string) => {
+      if (searchParam.length < 5) {
+        console.log("type more...")
+        setResult([])
+        return
+      }
+
+      try {
+        const result = await fetchSuggestions(searchParam, selectedParam)
+
+        console.log({ result })
+        setResult(result.suggestions)
+      } catch (e) {
+        console.error(e)
+      }
+    },
+    [fetchSuggestions]
+  )
+
+  // debounce requests
+  const debouncedFetchForAutocomplete = useDebounce({
+    // high debounce for testing to not eat up our credits
+    delay: 700,
+    callback: fetchForAutocomplete,
+  })
+
+  const buildAddressText = (suggestion): string => {
+    let whiteSpace = ""
+    let secondaryExtraInformation = ""
+    if (suggestion.secondary) {
+      if (suggestion.entries > 1) {
+        secondaryExtraInformation = " (" + suggestion.entries + " entries)"
+      }
+      whiteSpace = " "
+    }
+    return (
+      suggestion.street_line +
+      whiteSpace +
+      suggestion.secondary +
+      secondaryExtraInformation +
+      " " +
+      suggestion.city +
+      ", " +
+      suggestion.state +
+      " " +
+      suggestion.zipcode
+    )
+  }
```

Comment:
easy refactor, adapted from [buggy code example in the smarty docs.](https://www.smarty.com/docs/cloud/us-autocomplete-pro-api#pro-http-response-status)

Reply:
There should also be a comma before the city. Something like 
```ts
const buildingAddress = suggestion.street_line
if (suggestion.secondary) buildingAddress += ` ${suggestion.secondary}`

return [`${buildingAddress}, ${suggestion.city}`, suggestion.state, suggestion.zipcode].join(" ")
```